LandSandBoat/server
# [sql][lua]Add 108-Knot Quipu item, add missing despoiled items to items.lua global

**_I affirm:_**

## What does this pull request do?

- Adds 108-Knot Quipu item to relevant mobs (mobs/drop rate per FFXIDB)
  - Garlaige Citadel [S] and Sauromugue Champaign [S]
    - **Yagudo Abbot** - MobIDs `17178829`, `17178839`, `17178846`, `17448964`, `17448983`, `17449025`
    - **Yagudo Lutenist** - MobIDs `17178834`, `17178845`, `17178849`, `17448976`, `17449001`, `17449020`
    - **Yagudo Missionary** - MobIDs `17178830`, `17178840`, `17178843`, `17448973`, `17449002`, `17449019`
    - **Yagudo Prior** - MobIDs `17178832`, `17178838`, `17178847`, `17448969`, `17448984`, `17449026`
    - **Yagudo Pythoness** - MobIDs `17178835`, `17178841`, `17178850`, `17448977`, `17449003`, `17449023`
    - **Yagudo Zealot** - MobIDs `17178831`, `17178837`, `17178852`, `17448962`, `17449005`, `17449021`
  - Garlaige Citadel [S] only
    - **Yagudo Templar** - MobIDs `17448971`, `17449000`, `17449022`
  - Sauromugue Champaign [S] only
    - **Yagudo Inquisitor** - MobIDs `17178833`, `17178844`, `17178848`
- Re-organized, where possible, the drop lists the relevant mobs were a part of, either by grouping the relevent mobs together in a droplist or by grouping mobs NOT relevant to this item drop into different drop lists (but had the same drop items).
  - Audited all the drop lists (and converted to the usage of the drop percentage variables) that the mobs relevant to this item were included in **as well as** any non-relevant drop lists that were touched in this audit by the previously mentioned movements of the relevant mobs out of them.
- Added 3 new droplists where no suitable re-organization of drop lists was possible.
- Updated drop list IDs in *mob_groups.sql* of relevant/non-relevant mobs to match movements made in *mob_droplist.sql*.
- Added ROAST_CARP and ROAST_PIPIRA despoiled items to items.lua global file (items were discovered missing from the global file during the testing phase of the above changes).
- Typo correction in the **comment** of Wild Cudgel item (DropID 1338) in *mob_droplist.sql* (I randomly passed by it during my work on this PR and discovered the typo). 
  - No DropID data changes, just a correction of the comment.

## Steps to test these changes

- Prior to pulling down this PR, unmercilessly slaughter the above mobs that should drop the 108-Knot Quipu item and kill them as many times as you want, they will _**knot**_ ever give you the item, even if you kill them more or less than **108** times....just **qui**t whenever you are satisfied they aren't gonna drop it, **p**ick **u**p your gear, head here, and pull down this PR.....

- After pulling down this PR, give these above indicated mobs one more chance and kill any of them for the 108-Knot Quipu item (disclaimer that it MAY take more than ONE chance for the mobs to drop said item, depending on your Treasure Hunter level and what side of the bed RNGesus woke up on) (usage of the ``!respawnmob`` command is suggested).

I can provide audit criteria for the drop rates used if needed, not going to list all the criteria in this part of the ticket.

EDIT (03/11/2023) - Added MobID goodness for all of the relevant mobs that drop the 108-Knot Quipu item.
